components/login-register/register/pages/adding-teams-page/modals/teams-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { motion } from 'framer-motion'
+import { useEffect, useMemo, useState } from 'react'
+
+import Modal from '@/components/common/Modal/modal'
+import Spinner from '@/components/common/Spinner/spinner'
+import { ToggleSwitch } from '@/components/login-register/common/Input/toggleSwitch'
+import { NothingMessage } from '@/components/login-register/common/NothingMessage/nothing-message'
+import { SearchInput } from '@/components/Navbar/common/search-input'
+import ArrowLeftIcon from '@/icons/register/ArrowLeftIcon.svg'
+import EmptyTeamsIllustration from '@/icons/register/empty-state-team-search.svg'
+
+import { useTeamsList } from '../../../hooks'
+import { ExpandedButton } from '../common/expanded-button'
+import { getFormattedResult, getGenderValue } from '../helpers'
+import { useDebounce } from '../hooks'
+import { ITeamsModal } from '../types'
+
+export const TeamsModal = ({
+  openModal,
+  closeModal,
+  setSelected,
+}: ITeamsModal) => {
+  const queryClient = useQueryClient()
+  const [searchValue, setSearchValue] = useState('')
+  const [gender, setGender] = useState('Męskie')
+  const debouncedValue = useDebounce(searchValue, 400)
```

Comment:
if I'm guessing correctly
```suggestion
  const debouncedSearchValue = useDebounce(searchValue, 400)
```